### PR TITLE
Add unit tests for Curve3D

### PR DESCRIPTION
**Pull request by brno32**
Most of this is just the same code from https://github.com/godotengine/godot/pull/71308 but modified for 3 dimensions (Vector2 -> Vector3). I do not mean to pass off all of this work as my own, as the `Curve2D` test was my guide.

Seemed like an easy task to help get myself familiarized. All of the public methods from this class are called in the test at least once now I believe.

Should check off the `Curve3D` class from https://github.com/godotengine/godot/issues/43440

This was a fun way to try and wrap my head around Bézier curves!

### CODE REVIEW

File: tests/scene/test_curve_3d.h [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+TEST_CASE("[Curve3D] Even length tessellation") {
+	Ref<Curve3D> curve = memnew(Curve3D);
+	add_sample_curve_points(curve);
+
+	const int default_size = curve->tessellate_even_length().size();
+
+	// Default tessellate_even_length tolerance_length is 20.0, by adding a 100 units
+	// straight, we expect the total size to be increased by more than 5,
+	// that is, the algo will pick a length < 20.0 and will divide the straight as
+	// well as the curve as opposed to tessellate() which only adds the final point
```
Comment (Geometror) [inline tests/scene/test_curve_3d.h:254]:
```suggestion
	// well as the curve as opposed to tessellate() which only adds the final point.
```
Just a small nitpick.
Maybe split up the sentence since it's quite long and hard to follow.

**Review by Geometror — APPROVED** [review]
Great work!

**Comment by akien-mga** [comment]
Thanks!
